src/advanced_inv.cpp

```diff
@@ -839,24 +844,27 @@ bool advanced_inventory::move_all_items( bool nested_call )
         drop_locations dropped_favorite;
 
         if( spane.get_area() == AIM_INVENTORY ) {
-            for( size_t index = 0; index < g->u.inv.size(); ++index ) {
-                const std::list<item> &stack = g->u.inv.const_stack( index );
-                const item &it = stack.front();
-                item_location indexed_item( g->u, const_cast<item *>( &it ) );
-
-                if( !spane.is_filtered( it ) ) {
-                    int count;
-                    if( it.count_by_charges() )                         {
-                        count = it.charges;
-                    } else {
-                        count = stack.size();
+            //add all solid top level items
+            for( item &cloth :  g->u.worn ) {
+                for( item *it : cloth.contents.all_items_top() ) {
+                    if( !it->made_of_from_type( phase_id::SOLID ) ) {
+                        continue;
                     }
-                    if( it.is_favorite ) {
-                        dropped_favorite.emplace_back( indexed_item, count );
-                    } else {
-                        dropped.emplace_back( indexed_item, count );
+                    if( !spane.is_filtered( *it ) ) {
+                        int count;
```

Comment:
```suggestion
                        const int count = it->count();
```
`item::count` does the same as your code.